crates/rattler_networking/tests/integration_test.rs

```diff
@@ -0,0 +1,220 @@
+        match self.handle.kill() {
+            Ok(_) => {}
+            Err(e) => eprintln!("Failed to kill Minio server: {e}"),
+        }
+    }
+}
+
+fn run_subprocess(cmd: &str, args: &[&str], env: &HashMap<&str, &str>) -> std::process::Output {
+    let mut command = std::process::Command::new(cmd);
+    command.args(args);
+    for (key, value) in env {
+        command.env(key, value);
+    }
+    let output = command.output().expect("Failed to run command");
+    assert!(output.status.success());
+    output
+}
+
+fn init_channel() {
+    let env = &HashMap::from([(
+        "MC_HOST_local",
+        "http://minioadmin:minioadmin@localhost:9000",
+    )]);
+    run_subprocess("mc", &["mb", "local/rattler-s3-testing"], env);
+    run_subprocess(
+        "mc",
+        &[
+            "cp",
+            PathBuf::from("../../test-data/test-server/repo/noarch/repodata.json")
+                .to_str()
+                .unwrap(),
+            "local/rattler-s3-testing/my-channel/noarch/repodata.json",
+        ],
+        env,
+    );
+    run_subprocess(
+        "mc",
+        &[
+            "cp",
+            PathBuf::from("../../test-data/test-server/repo/noarch/test-package-0.1-0.tar.bz2")
+                .to_str()
+                .unwrap(),
+            "local/rattler-s3-testing/my-channel/noarch/test-package-0.1-0.tar.bz2",
+        ],
+        env,
+    );
+}
+
+async fn with_env(
+    env: HashMap<&str, &str>,
+    f: impl FnOnce() -> std::pin::Pin<Box<dyn std::future::Future<Output = ()> + Send>>,
+) {
+    for (key, value) in &env {
+        std::env::set_var(key, value);
+    }
+    f().await;
+    for (key, _) in env {
+        std::env::remove_var(key);
+    }
+}
+
+/* ------------------------------------ FIXTURES ------------------------------------ */
+
+#[fixture]
+fn minio_server() -> MinioServer {
+    let server = MinioServer::new();
+    init_channel();
+    server
+}
+
+#[fixture]
+fn auth_file() -> (TempDir, std::path::PathBuf) {
+    let temp_dir = tempdir().unwrap();
+    let aws_config = r#"
+{
+    "s3://rattler-s3-testing/my-channel": {
+        "S3Credentials": {
+            "access_key_id": "minioadmin",
+            "secret_access_key": "minioadmin"
+        }
+    }
+}
+"#;
+    let credentials_path = temp_dir.path().join("credentials.json");
+    std::fs::write(&credentials_path, aws_config).unwrap();
+    (temp_dir, credentials_path)
+}
+
+#[fixture]
+fn aws_config() -> (TempDir, std::path::PathBuf) {
+    let temp_dir = tempdir().unwrap();
+    let aws_config = r#"
+[profile default]
+aws_access_key_id = minioadmin
+aws_secret_access_key = minioadmin
+endpoint_url = http://localhost:9000
+region = eu-central-1
+"#;
+    let aws_config_path = temp_dir.path().join("aws.config");
+    std::fs::write(&aws_config_path, aws_config).unwrap();
+    (temp_dir, aws_config_path)
+}
+
+/* -------------------------------------- TESTS ------------------------------------- */
```

Comment:
we should add a test for public buckets here as wll (both using `S3Config::FromAWS` and `S3Config::Custom`

Reply:
Done